File: src/main/java/com/snooper/service/CommandHandler.java
package com.snooper.service;

import com.snooper.constant.Action;
import com.snooper.dto.Plateau;
import com.snooper.dto.RoboticRover;

import java.util.List;

public interface CommandHandler {
    Plateau parsePlateau(String command);

    RoboticRover parseRoboticRover(String command);

    List<Action> parseActions(String command);
}

File: src/main/java/com/snooper/service/CommandHandler.kt
package com.snooper.service

import com.snooper.constant.Action
import com.snooper.dto.Plateau
import com.snooper.dto.RoboticRover

interface CommandHandler {
    fun parsePlateau(command: String): Plateau

    fun parseRoboticRover(command: String): RoboticRover

    fun parseActions(command: String): List<Action>
}